scripts/flow_chart.py

```diff
@@ -0,0 +1,138 @@
+
+    def get_localized_name(self, item_id, dictionary):
+        """Get English localized name from dictionary, fallback to item_id if not found."""
+        try:
+            return dictionary[item_id]['localized_name']['en']
+        except KeyError:
+            return item_id
+
+    def collect_recipe_variants(self):
+        """Collect all recipe variants with non-zero solution values."""
+        variants = defaultdict(list)
+        
+        for recipe_var in self.solver_recipes.values():
+            if recipe_var.solution_value() > 1e-9:
+                data = parse_recipe_id(recipe_var.name())
+                data['solution_value'] = recipe_var.solution_value()
+                
+                recipe_name = data['recipe_name']
+                if data['machine'] == 'recycler':
+                    recipe_name = recipe_name.replace('-recycling', '')
+                    
+                variants[recipe_name].append(data)
+                
+        return variants
+
+    def format_modules(self, num_qual, num_prod):
+        """Format module string (e.g., '2Q1P' for 2 quality and 1 production modules)."""
+        mods = []
+        if int(num_qual):
+            mods.append(f'{num_qual}Q')
+        if int(num_prod):
+            mods.append(f'{num_prod}P')
+        return ' ' + ''.join(mods) if mods else ''
+
+    def get_node_ids(self, recipe, variant):
+        """Generate graph and class IDs for a variant."""
+        graph_id = f'{recipe}_{variant["machine"]}_{variant["recipe_quality"]}'
+        class_id = variant["recipe_quality"]
+        
+        if variant['machine'] == 'recycler':
+            class_id += '-recycling'
+            
+        return graph_id, class_id
+
+    def render_variant(self, recipe, variant_list, classes):
+        """Render a single recipe variant as a Mermaid subgraph."""
+        recipe_name = self.get_localized_name(recipe, self.recipes)
+        output_lines = []
+        
+        for variant in variant_list:
+            # Get localized names
+            machine = self.get_localized_name(variant['machine'], self.items)
+            name = self.get_localized_name(variant['recipe_name'], self.recipes)
+            
+            # Format modules string
+            mods = self.format_modules(variant['num_qual_modules'], 
+                                     variant['num_prod_modules'])
+            
+            # Generate IDs and track classes
+            graph_id, class_id = self.get_node_ids(recipe, variant)
+            classes[class_id].append(graph_id)
+            
+            # Create node definition
+            node_text = (f'{graph_id}[{name.replace("(","").replace(")", "")} - '
+                        f'{variant["recipe_quality"]} - {machine}{mods} x '
+                        f'{math.ceil(variant["solution_value"])}]')
+            output_lines.append(node_text)
+            
+        return f"subgraph {recipe_name}\n{'\n'.join(output_lines)}\nend\n"
```

Comment:
I got an error when running this line, `SyntaxError: f-string expression part cannot include a backslash`. Apparently using `\n` inside an f-string curly-brace expression is forbidden in certain versions of python (I'm using 3.11.8). Can you refactor this line so the `'\n'.join(output_lines)` happens before the curly braces?